### PR TITLE
Wallet API: support integrated addresses in address book.

### PR DESCRIPTION
**Pull request by Jaqueeee**
Fixes https://github.com/monero-project/monero-core/issues/392

### CODE REVIEW

File: src/wallet/api/address_book.cpp [hunk 1]
```diff
@@ -43,30 +43,43 @@ AddressBook::~AddressBook() {}
...
     m_errorCode = Invalid_Address;
     return false;
   }
 
-  crypto::hash pid32 = cryptonote::null_hash;
-  bool long_pid =  (payment_id.empty())? false : tools::wallet2::parse_long_payment_id(payment_id, pid32);
-  if(!payment_id.empty() && !long_pid) {
+  crypto::hash payment_id = cryptonote::null_hash;
+  bool has_long_pid = (payment_id_str.empty())? false : tools::wallet2::parse_long_payment_id(payment_id_str, payment_id);
+
```
Comment (MoroccanMalinois) [inline src/wallet/api/address_book.cpp]: Adding entry with simple address + short payment id doesn't work in address book (GUI).
Based on simplewallet.cpp L3561 : should add "parse_short_payment_id" if "parse_long_payment_id" fails

Comment (Jaqueeee) [inline src/wallet/api/address_book.cpp]: It's intentional. The short payment ID should only be used in an integrated address.

Comment (MoroccanMalinois) [inline src/wallet/api/address_book.cpp]: OK Thx Jacqueee. The message "Invalid Payment ID" make it seems like it's a bug. Maybe a more specific message "Short payment ID should only be used in an integrated address" ?

Comment (Jaqueeee) [inline src/wallet/api/address_book.cpp]: Good idea. I'll change that.

**Comment by Jaqueeee** [comment]
Rebased and edited error msg

**Pull request merged**